arch/mx.rst

```diff
@@ -1,13 +1,10 @@
 .. _mx:
 
-Masterless Mode (beta)
-======================
+Masterless Mode
```

Comment:
See below (since I can't comment on unchanged lines). In "Limitations compared to Citus" we note 7.4. Can we update that to 8.0? Also, in Onder's email titled "Highlights from Citus 8.0: Citus MX" , see section: "Limitations/Differences of Citus MX compared to single-coordinator Citus". That section is a bit detailed, so we'll need to condense that. But we should update the limitations section with this content.

Reply:
I'll take care of the version updates in the release checklist PR, and the MX limitations in this one.

Reply:
Added a concise list of limitations